azkaban-common/src/main/java/azkaban/executor/ExecutionControllerUtils.java

```diff
@@ -318,7 +318,14 @@ public static String createJobLinkUrl(final ExecutableFlow exFlow, final String
       jobLinkUrl = url.toString();
     } else {
       // If RM job link is invalid or has expired, fetch the job link from JHS or SHS.
-      if (exFlow.getExecutableNode(jobId).getType().equals(SPARK_JOB_TYPE)) {
+      final ExecutableNode node = exFlow.getExecutableNodePath(jobId);
+      if (node == null) {
```

Comment:
In which scenario this is a possibility?

Reply:
It should not be null unless the input jobId parameter is incorrect.

Reply:
Thanks